### PR TITLE
Repository name duplicate verification request debounce & fetch gitignore data from server

### PR DESCRIPTION
**Pull request by bentshrimp**
## Describe changes

- verificate repoName
- request for repoName duplicate verfication debounce
- fetch gitignore data from server

## Types of changes

What is the type of code change?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (changes that resolve errors)
- [x] New feature (changes which adds functionality)
- [ ] Breaking change (big changes that affect existing functionality)
- [ ] Documentation Update

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the "README.md"
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

_Please let me know if there's anything else to explain_

### CODE REVIEW

File: src/components/step1/RequiredFieldContainer.js [hunk 1]
```diff
@@ -69,23 +104,22 @@ export const RequiredFieldContainer = () => {
...
-      setValidateCheck(response.data);
-      if (response.data) {
-        setHelperText("checked");
-      } else {
-        setHelperText("error");
-      }
+      return response.data;
     } catch (e) {
       console.error(e);
       alert("기록 시작 실패. 재시도해주세요.");
```
Comment (sinji2102) [inline src/components/step1/RequiredFieldContainer.js:110]: Please use English

**Review by sinji2102 — COMMENTED** [review]
👍🏻

File: src/components/step1/RequiredFieldContainer.js (hunk 1 above)
Comment (bentshrimp) [reply src/components/step1/RequiredFieldContainer.js:110]: Got it i'll fix that

**Review by hummingbbird — APPROVED** [review]
👍

**Pull request merged**